domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/review/service/ReviewService.kt

```diff
@@ -85,8 +113,16 @@ class ReviewService(
    * is created with a status of SCHEDULED. This therefore becomes their latest Review Schedule.
    *
    * If the prisoner does not already have an active [ReviewSchedule] a [ReviewScheduleNotFoundException] is thrown.
+   *
+   * Only certain sentence types are supported to be able to create a new [ReviewSchedule], and of those SENTENCED and
+   * RECALL require the prisoner to have a release date. Throws a subclass of [InvalidReviewScheduleException] if these
+   * validation rules are not met.
    */
   fun createReview(createCompletedReviewDto: CreateCompletedReviewDto): CompletedReviewDto {
+    with(createCompletedReviewDto) {
```

Comment:
Validate at the public API methods before we do anything, with one of the new exceptions being thrown if validation fails